File: Mathematics/Quadratic_equation_root.java
class Solution {
    public ArrayList<Integer> quadraticRoots(int a, int b, int c) {
        // code here
        ArrayList<Integer> d = new ArrayList<Integer>();
        double det = ( (b*b) - (4.0 * a * c) );
        if( det < 0 ) { // check if no roots exist
            d.add(-1);
            return d;
        }
        det = Math.sqrt(det);
        double detPlus = Math.floor((-b+det)/(2.0*a));
        double detMinus = Math.floor((-b-det)/(2.0*a));
        if(detPlus < detMinus) {
            d.add((int)detMinus);
            d.add((int)detPlus);
        } else {
            d.add((int)detPlus);
            d.add((int)detMinus);
        }
        return d;
    }
}

File: Mathematics/Quadratic_equation_root.cpp
class Solution {
  public:
    vector<int> quadraticRoots(int a, int b, int c) {
        vector<int> roots;
        double det = (b * b) - (4.0 * a * c);
        if (det < 0) { // check if no roots exist
            roots.push_back(-1);
            return roots;
        }
        det = sqrt(det);
        double detPlus = floor((-b + det) / (2.0 * a));
        double detMinus = floor((-b - det) / (2.0 * a));
        if (detPlus < detMinus) {
            roots.push_back((int)detMinus);
            roots.push_back((int)detPlus);
        } else {
            roots.push_back((int)detPlus);
            roots.push_back((int)detMinus);
        }
        return roots;
    }
};
